src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketBuildStatus.java

```diff
@@ -72,7 +72,7 @@ public BitbucketBuildStatus(String hash, String description, String state, Strin
         this.description = description;
         this.state = state;
         this.url = url;
-        this.key = key;
+        this.key = key.substring(0, Math.min(key.length(), 40));
```

Comment:
I would suggest fixing this the same way done in https://issues.jenkins-ci.org/browse/JENKINS-32781

I had logged a new Jira for this not realizing this was already out there (see https://issues.jenkins-ci.org/browse/JENKINS-38146)